src/util/address.rs

```diff
@@ -13,32 +13,8 @@
 
 //! Addresses
 //!
-//! Support for ordinary base58 Bitcoin addresses and private keys
+//! Support for various types of Bitcoin addresses
 //!
-//! # Example: creating a new address from a randomly-generated key pair
```

Comment:
Why remove the example?

Reply:
because we have no random generator available in the project

Reply:
secp is compiled such that it has only weak pseudo random generators.

Reply:
examples with //! are compiled and fail in this case

Reply:
> secp is compiled such that it has only weak pseudo random generators.

how so?

> examples with //! are compiled and fail in this case

This example doesn't fail for me, including the tests with `thread_rng`.

Reply:
it does fail if you have a clean checkout as with travis. The reason is that secp256k1 0.12 does not have the feature std set which makes thread_rng disappear. This does not compile as is, but would compile again with secp 0.15

Reply:
actually this is with secp 0.15, so my above explain is wrong. Still it does not build with travis if thread_rng is used...

Reply:
that is why I had oll the other changes. This example is BTW also on wrong place as this is the Address class and is not about key generation.

Reply:
I did a clean build of this branch locally and inserted a 
    use secp256k1::rand::thread_rng;
and promtly got an
error[E0432]: unresolved import `secp256k1::rand::thread_rng`

so thread_rng is somehow disabled although I can not et ecplain why.

Reply:
it seems that
```
[dependencies.secp256k1]
version = "0.15"
features = [ "rand" ]
```
in rust-bitcoin's Cargo.toml
overrides
```
[features]
unstable = []
default = ["std"]
```
in rust-secp256k1

Reply:
and not having std feature disables thread_rng. 
@apoelstra should tell us if not having std in rust-bitcoin's Cargo is accidental or not.

Reply:
Ah okay, after switching from rust 1.22 to 1.36 I'm able to reproduce the missing thread_rng. 

Having working examples demonstrating the usage of cryptographically secure PRNGs is quite valuable. And also since rust-bitcoin isn't no-std I don't see a reason to not enable std when importing rust-secp.

Reply:
@tamasblummer it is an accident that `std` is missing from rust-bitcoin's Cargo. Good catch!

Reply:
It's because rust-bitcoin's Cargo.toml predates secp having the `std` feature (which doesn't even exist in rust-secp 0.12)

Reply:
@apoelstra a further problem is that rust-secp256k1 defines:
```
[dependencies.rand]
version = "0.6"
optional = true
default-features = false
```
which according to rand documentation means std is disabled in rand, and therefore no thread_rng: https://crates.io/crates/rand
```
Rand supports limited functionality in no_std mode (enabled via default-features = false). In this case, OsRng and from_entropy are unavailable (unless getrandom is enabled), large parts of seq are unavailable (unless alloc is enabled), and thread_rng and random are unavailable.
```

Reply:
cc @elichai can you PR to rust-secp to make the `std` feature propagate to `rand`?

Reply:
Hmm. I actually don't like how cargo works in that sense.
The way cargo does features mean that if there's this dependency tree `A->B->C` and `A` want to use something with a feature in `C` either `B` needs to enable it forcing everyone to compile some more code.
or the user can also import `C` with that feature enable, but now the user need to maintain the `C` dependency and make sure it matches the same version of `C` as in `B` :\

Reply:
Yes, cargo assumes that features are additive (which often isn't true, but whatever..) and that it can therefore enable features on a crate whenever any dependency asks for it.

In this case I think it's the right thing though ... if you're enabling `std` on rust-secp then why wouldn't you want `std` on its dependencies? You've already lost the ability to compile without libstd. (Sure, it's "extra code", but Rust is good at pruning unused stuff out of final binaries.)